### DISCUSSION TITLE
Sync scene instantiation in 5.0.0 (which is obsolete) results in unresponsive Editor

### DISCUSSION THREAD

**Discussion by nikeruga** [atteneder/glTFast#544] [General]
hi.
my project.
draco 4.0.0
gltFast 4.9.1

this version is glb load ok.
but change gltFast 5.0.
glb load has unity sleep.
![2022-12-16 21 04 48](https://user-images.githubusercontent.com/39543900/208164319-b6090f0c-d8c4-47e4-870f-2de8a0e1b1cb.jpg)

and update unity main version 2021.3.16f1
same error..

my code very simple..

        var gltf = new GLTFast.GltfImport();
        var success = await gltf.Load(gltfFile);
        if(success)
        {
            success = gltf.InstantiateMainScene(modelMainObject.transform);
        }
        else
        {
            Debug.LogError("Loading glTF failed!");
        }

**Comment by atteneder** [comment]
I assume the issue disappeared and you wanted to delete the discussion?

Also, for errors, bugs or crashes please create an Issue, not a discussion, and provide the glTF file that causes the error.

**Reply by nikeruga** [reply]
The problem is the same.
When updating to 5.0 and loading, the unity program stops.

At first, I thought it was because of draco, but when I changed to version 4.9.1, there was no problem.


![2022-12-16 21 04 48](https://user-images.githubusercontent.com/39543900/208163939-b1e65576-a1cc-4d44-8cd4-7da9fddd1118.jpg)

**Reply by atteneder** [reply]
The only value I get from the video is validation that the problem actually exists. That I never questioned though.

Again, much more helpful would be to get the glTF asset that causes the problem (pretty please). I have a suspicion about what causes the problem, but without proper reproduction one can only guess the solution.

**Reply by atteneder** [reply]
I second the solution proposed by @Merichbier . Skip the obsolete instantiation and call the new, async variant:

```csharp
success = await gltf.InstantiateMainSceneAsync(modelMainObject.transform);
```

**Comment by Merichbier** [comment]
I've had similar issue today with 5.0 after upgrade from 4.9.1.
To fix it, I move from sync method InstantiateScene to the async InstantiateSceneAsync and it worked great.
It is specified in the upgrade file FYI.

**Comment by nikeruga** [comment]
made recording my error screen.!

https://drive.google.com/file/d/18X2FnaqOk1tNSJaOtjAdFDNciGu-u1LC/view?usp=sharing
